src/screens/ContactScreen/index.js

```diff
@@ -45,17 +46,19 @@ const {width} = Dimensions.get('screen');
 const ContactScreen = ({navigation}) => {
   const [loading, setLoading] = React.useState(false);
   const [users, setUsers] = React.useState([]);
+  const [usersSearch, setUsersSearch] = React.useState([]);
   const [profile] = React.useContext(Context).profile;
   const [isRecyclerViewShown, setIsRecyclerViewShown] = React.useState(false);
+  const [isRecyclerViewShownSearch, setIsRecyclerViewShownSearch] = React.useState(false);
   const [layoutProvider, setLayoutProvider] = React.useState(() => {});
-  const [refreshing, setRefreshing] = React.useState(false);
   const [dataProvider, setDataProvider] = React.useState(null);
+  const [layoutProviderSearch, setLayoutProviderSearch] = React.useState(() => {});
+  const [dataProviderSearch, setDataProviderSearch] = React.useState(null);
+  const [refreshing, setRefreshing] = React.useState(false);
   const [text, setText] = React.useState('');
-  const [debouncedText, setDebouncedText] = React.useState('');
   const [followed, setFollowed] = React.useState([profile.myProfile.user_id]);
   const [usernames, setUsernames] = React.useState([profile.myProfile.username]);
   const [selectedUsers, setSelectedUsers] = React.useState([]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new state variables `usersSearch`, `isRecyclerViewShownSearch`, `layoutProviderSearch`, `dataProviderSearch` are introduced. These seem to be related to the search functionality. However, it's not clear why we need separate state variables for search and non-search scenarios. If these states are only used in the context of a search operation, consider encapsulating them within a custom hook or context to improve modularity and maintainability.



<!-- This is an auto-generated comment by OSS CodeRabbit -->